screens/TrackDetailsScreen.js

```diff
@@ -0,0 +1,131 @@
+import React, { Component } from 'react';
+import {
+    View,
+    Text,
+    StyleSheet,
+    ScrollView,
+} from 'react-native';
+import LyricsProvider from "../provider/LyricsProvider"
+
+export default class SearchResultScreen extends Component {
+    state = {
+        trackName: 'Unknown',
+        trackID: '123',
+        details: {
+            artist: "Jane Doe",
+            album: "The Album",
+            releaseDate: "2010-10-10",
+            explicit: "No"
+        },
+        lyrics: "Not found",
+        loading: true,
+    }
+    
+    componentDidMount() {
+        let { trackName } = this.props.route.params;
+        let { trackID } = this.props.route.params;
+        let { details } = this.props.route.params;
+        if(details.explicit == 0) {
+            details.explicit = "No";
+        }
+        else {
+            details.explicit = "Yes";
+        }
+        this.setState({ trackName, trackID, details });
```

Comment:
Algumas sugestões das linhas 11 a 34:

- Melhor utilizar um constructor nesse caso, algo como
```
constructor(props) {
    this.state = {
        trackName: this.props.route.params.trackName,
        trackID: this.props.route.params.trackId,
        (etc etc)
    }
}
```
O ponto é o seguinte: o constructor(props) é chamado antes do componentDidMount(), e as props já estão disponíveis no constructor, então faz mais sentido setar tudo que for possível no constructor.